### ISSUE TITLE
`publishJacocoCodeCoverageToBitbucket` wants the JaCoCo report tasks executed in the same Gradle build execution

### ISSUE THREAD

**Issue by rkrisztian** [marcelkliemannel/gradle-bitbucket-code-coverage-plugin#1]
I noticed [this code](https://github.com/marcelkliemannel/gradle-bitbucket-code-coverage-plugin/blob/main/src/main/kotlin/dev/turingcomplete/bitbucketcodecoverage/BitbucketCodeCoveragePlugin.kt#L69C1-L69C69):

```kotlin
      task.onlyIf { jacocoReportTasks.any { task -> task.didWork } }
```

However, according to [the API docs](https://docs.gradle.org/current/javadoc/org/gradle/api/Task.html#getDidWork--), `didWork` may return true even if the task was run in a previous Gradle execution, but after when running the publication task separately. So in other words, this forces us to execute the publication tasks in the same run as the tasks that execute the tests and generate coverage.

In other words only this works, which is not so nice:

```bash
gw check
gw --rerun-tasks publishJacocoCodeCoverageToBitbucket
```

**Comment by marcelkliemannel** [comment]
Version 1.1.0 has now a property `onlyRunIfJacocoReportTasksDidWork` to disable this dependenvy.
